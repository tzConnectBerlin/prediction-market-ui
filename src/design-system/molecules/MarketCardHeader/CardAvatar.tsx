import { Identicon, IdenticonProps } from '../../atoms/Identicon';

export interface CardAvatarProps {
  title?: string;
  /**
   * market ipfs hash
   */
  hash?: string;
  /**
   * Icon url to use
   */
  iconURL?: string;

  iconSize?: IdenticonProps['iconSize'];

  iconType?: IdenticonProps['type'];
}

export const CardAvatar: React.FC<CardAvatarProps> = ({
  title,
  hash,
  iconURL,
  iconSize = 'xl',
  iconType = 'blockies',
}) => {
  return <Identicon seed={hash ?? title ?? ''} url={iconURL} iconSize={iconSize} type={iconType} />;
};
