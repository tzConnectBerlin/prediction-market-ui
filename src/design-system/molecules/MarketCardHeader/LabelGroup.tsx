import React from 'react';
import { Grid, useMediaQuery, useTheme } from '@material-ui/core';
import { GiAlarmClock } from 'react-icons/gi';
import { Label, LabelProps } from '../../atoms/Label';

type CardStateProps = Omit<LabelProps, 'text'>;

export interface LabelGroupProps {
  /**
   * card label mostly used for market steps (Market,Auction,...)
   */
  cardState: string;
  /**
   * market close timestamp to display on the card
   */
  closeDate: string;
  cardStateProps?: CardStateProps;
}

export const LabelGroup: React.FC<LabelGroupProps> = ({ cardState, closeDate, cardStateProps }) => {
  const theme = useTheme();
  const isMobile = useMediaQuery(theme.breakpoints.down('sm'));

  return (
    <Grid container direction="row" spacing={1} justifyContent={isMobile ? 'center' : 'inherit'}>
      <Grid item>
        <Label text={cardState} {...cardStateProps} />
      </Grid>
      <Grid item>
        <Label
          text={closeDate}
          backgroundColor={theme.palette.grey[500]}
          fontColor={theme.palette.text.primary}
          icon={<GiAlarmClock />}
        />
      </Grid>
    </Grid>
  );
};
