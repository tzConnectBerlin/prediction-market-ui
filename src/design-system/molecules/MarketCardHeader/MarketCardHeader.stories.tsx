import { Story, Meta } from '@storybook/react/types-6-0';
import { MarketCardHeader, MarketCardHeaderProps } from './MarketCardHeader';

export default {
  title: 'Molecules/MarketCardHeader',
  component: MarketCardHeader,
} as Meta;

const Template: Story<MarketCardHeaderProps> = (args) => <MarketCardHeader {...args} />;

export const WithMarketCloseDate = Template.bind({});
WithMarketCloseDate.args = {
  iconURL: 'https://w.wallhaven.cc/full/vg/wallhaven-vg7lv3.jpg',
  title: 'Market card with image icon',
  closeDate: 'Closed',
};

export const WithLongTitleText = Template.bind({});
WithLongTitleText.args = {
  iconURL: 'https://w.wallhaven.cc/full/vg/wallhaven-vg7lv3.jpg',
  title:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
  closeDate: '6th May 2021 09:32',
};

export const WithHashOnly = Template.bind({});
WithHashOnly.args = {
  hash: 'QmYgtfMBZo3ajW5rmUesVfHSJu5nT6fT3cRcvr2fpfbzo3',
  title: 'Market card with hash generated icon',
  cardState: 'Market',
  closeDate: '6th May 2021 09:32',
};
