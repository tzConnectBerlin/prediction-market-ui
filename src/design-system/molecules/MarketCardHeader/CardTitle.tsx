import * as React from 'react';
import styled from '@emotion/styled';
import { Typography, TypographyProps } from '../../atoms/Typography';
import { LabelGroup, LabelGroupProps } from './LabelGroup';

export interface CardTitleProps extends LabelGroupProps {
  /**
   * market question
   */
  title: string;
  titleSize?: TypographyProps['size'];
}

const StyledTitle = styled.div`
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  overflow: hidden;
  padding: 0.3em 0;
`;

/**
 * TODO: Move to a separate folder
 *
 */

const defaultTitleSize = 'h2';

export const CardTitle: React.FC<CardTitleProps> = ({
  title,
  titleSize = defaultTitleSize,
  cardStateProps,
  ...rest
}) => (
  <>
    <LabelGroup {...rest} cardStateProps={cardStateProps} />
    <StyledTitle>
      <Typography size={titleSize} fontWeight="bold">
        {title}
      </Typography>
    </StyledTitle>
  </>
);
