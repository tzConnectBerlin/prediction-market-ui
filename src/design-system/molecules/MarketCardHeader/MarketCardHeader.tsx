import React from 'react';
import { CardHeader } from '@material-ui/core';
import { CardTitle, CardTitleProps } from './CardTitle';
import { CardAvatar, CardAvatarProps } from './CardAvatar';

export type MarketCardHeaderProps = CardTitleProps & CardAvatarProps;

export const MarketCardHeader: React.FC<MarketCardHeaderProps> = ({
  title,
  hash,
  iconURL,
  iconSize = 'xl',
  iconType = 'blockies',
  ...rest
}) => {
  return (
    <CardHeader
      avatar={
        <CardAvatar
          iconURL={iconURL}
          iconSize={iconSize}
          iconType={iconType}
          title={title}
          hash={hash}
        />
      }
      title={<CardTitle title={title} {...rest} />}
    />
  );
};
