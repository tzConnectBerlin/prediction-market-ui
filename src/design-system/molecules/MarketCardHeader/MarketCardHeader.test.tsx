import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { MarketCardHeader } from './MarketCardHeader';

describe('Snapshot testing MarketCardHeader Component', () => {
  it('renders correctly with image icon', () => {
    const cardHeader = renderer
      .create(
        <MarketCardHeader
          cardState="Auction"
          title="Market card with image icon"
          iconURL="https://w.wallhaven.cc/full/vg/wallhaven-vg7lv3.jpg"
        />,
      )
      .toJSON();
    expect(cardHeader).toMatchSnapshot();
  });

  it('renders correctly with hash generated icon', () => {
    const cardHeader = renderer
      .create(
        <MarketCardHeader
          cardState="Auction"
          title="Market card with image icon"
          hash="QmYgtfMBZo3ajW5rmUesVfHSJu5nT6fT3cRcvr2fpfbzo3"
        />,
      )
      .toJSON();
    expect(cardHeader).toMatchSnapshot();
  });

  it('renders correctly with different card label', () => {
    const cardHeader = renderer
      .create(
        <MarketCardHeader
          title="Auction step"
          iconURL="https://w.wallhaven.cc/full/vg/wallhaven-vg7lv3.jpg"
          cardState="Auction"
        />,
      )
      .toJSON();
    expect(cardHeader).toMatchSnapshot();
  });
});

describe('Element testing MarketCardHeader Component', () => {
  it('render correctly with Auction label', async () => {
    const { getByText } = render(
      <MarketCardHeader
        title="Market title"
        cardState="Auction"
        iconURL="https://w.wallhaven.cc/full/vg/wallhaven-vg7lv3.jpg"
      />,
    );

    expect(getByText(/Auction/i)).toBeInTheDocument();
  });

  it('render correctly with Market title', async () => {
    const { getByText } = render(
      <MarketCardHeader
        title="Market Title"
        iconURL="https://w.wallhaven.cc/full/vg/wallhaven-vg7lv3.jpg"
        cardState="Market"
      />,
    );
    expect(getByText(/Market Title/i)).toBeInTheDocument();
  });
});
