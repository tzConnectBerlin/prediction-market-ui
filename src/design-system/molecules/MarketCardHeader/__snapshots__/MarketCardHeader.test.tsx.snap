// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing MarketCardHeader Component renders correctly with different card label 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 16px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  margin-right: 16px;
}

.emotion-2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 40px;
  height: 40px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-size: 1.25rem;
  line-height: 1;
  border-radius: 50%;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-2.hasBackground {
  background-color: rgba(29, 34, 39, 0.04);
}

.emotion-2.xs {
  width: 24px;
  height: 24px;
}

.emotion-2.sm {
  width: 32px;
  height: 32px;
}

.emotion-2.md {
  width: 40px;
  height: 40px;
}

.emotion-2.lg {
  width: 48px;
  height: 48px;
}

.emotion-2.xl {
  width: 64px;
  height: 64px;
}

.emotion-2.xxl {
  width: 80px;
  height: 80px;
}

.emotion-2.max {
  width: 144px;
  height: 144px;
}

.emotion-3 {
  width: 100%;
  height: 100%;
  text-align: center;
  object-fit: cover;
  color: transparent;
  text-indent: 10000px;
}

.emotion-4 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-5 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  display: block;
}

.emotion-6 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: calc(100% + 8px);
  margin-top: -8px;
  width: calc(100% + 8px);
  margin-left: -8px;
  -webkit-box-pack: inherit;
  -ms-flex-pack: inherit;
  -webkit-justify-content: inherit;
  justify-content: inherit;
}

.emotion-6>.MuiGrid-item {
  padding-top: 8px;
}

.emotion-6>.MuiGrid-item {
  padding-left: 8px;
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-8 {
  border-radius: 0.2em;
  padding: 0.3em 0.6em;
  display: inline-block;
  background-color: #1976d2;
  color: #fff;
}

.emotion-8.label--icon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-8.label--icon>* {
  margin-right: 0.3em;
}

.emotion-8.label--icon>svg {
  width: 0.8em;
  height: 0.8em;
}

.emotion-9 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1.5rem;
  line-height: 1.334;
  letter-spacing: 0em;
}

.emotion-9.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.emotion-10 {
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  overflow: hidden;
  padding: 0.3em 0;
}

.emotion-11 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 300;
  font-size: 3.75rem;
  line-height: 1.2;
  letter-spacing: -0.00833em;
  font-weight: 700;
}

.emotion-11.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

<div
  className="MuiCardHeader-root emotion-0"
>
  <div
    className="MuiCardHeader-avatar emotion-1"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circular xl hasBackground emotion-2"
    >
      <img
        className="MuiAvatar-img emotion-3"
        src="https://w.wallhaven.cc/full/vg/wallhaven-vg7lv3.jpg"
      />
    </div>
  </div>
  <div
    className="MuiCardHeader-content emotion-4"
  >
    <span
      className="MuiTypography-root MuiTypography-body2 MuiCardHeader-title emotion-5"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 emotion-6"
      >
        <div
          className="MuiGrid-root MuiGrid-item emotion-7"
        >
          <span
            className=" emotion-8"
          >
            <h5
              className="MuiTypography-root MuiTypography-h5  emotion-9"
              lines={3}
            >
              Auction
            </h5>
          </span>
        </div>
      </div>
      <div
        className="emotion-10"
      >
        <h2
          className="MuiTypography-root MuiTypography-h2  emotion-11"
          lines={3}
        >
          Auction step
        </h2>
      </div>
    </span>
  </div>
</div>
`;

exports[`Snapshot testing MarketCardHeader Component renders correctly with hash generated icon 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 16px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  margin-right: 16px;
}

.emotion-2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 40px;
  height: 40px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-size: 1.25rem;
  line-height: 1;
  border-radius: 50%;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-2.hasBackground {
  background-color: rgba(29, 34, 39, 0.04);
}

.emotion-2.xs {
  width: 24px;
  height: 24px;
}

.emotion-2.sm {
  width: 32px;
  height: 32px;
}

.emotion-2.md {
  width: 40px;
  height: 40px;
}

.emotion-2.lg {
  width: 48px;
  height: 48px;
}

.emotion-2.xl {
  width: 64px;
  height: 64px;
}

.emotion-2.xxl {
  width: 80px;
  height: 80px;
}

.emotion-2.max {
  width: 144px;
  height: 144px;
}

.emotion-3 {
  width: 100%;
  height: 100%;
  text-align: center;
  object-fit: cover;
  color: transparent;
  text-indent: 10000px;
}

.emotion-4 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-5 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  display: block;
}

.emotion-6 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: calc(100% + 8px);
  margin-top: -8px;
  width: calc(100% + 8px);
  margin-left: -8px;
  -webkit-box-pack: inherit;
  -ms-flex-pack: inherit;
  -webkit-justify-content: inherit;
  justify-content: inherit;
}

.emotion-6>.MuiGrid-item {
  padding-top: 8px;
}

.emotion-6>.MuiGrid-item {
  padding-left: 8px;
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-8 {
  border-radius: 0.2em;
  padding: 0.3em 0.6em;
  display: inline-block;
  background-color: #1976d2;
  color: #fff;
}

.emotion-8.label--icon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-8.label--icon>* {
  margin-right: 0.3em;
}

.emotion-8.label--icon>svg {
  width: 0.8em;
  height: 0.8em;
}

.emotion-9 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1.5rem;
  line-height: 1.334;
  letter-spacing: 0em;
}

.emotion-9.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.emotion-10 {
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  overflow: hidden;
  padding: 0.3em 0;
}

.emotion-11 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 300;
  font-size: 3.75rem;
  line-height: 1.2;
  letter-spacing: -0.00833em;
  font-weight: 700;
}

.emotion-11.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

<div
  className="MuiCardHeader-root emotion-0"
>
  <div
    className="MuiCardHeader-avatar emotion-1"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circular xl hasBackground emotion-2"
    >
      <img
        className="MuiAvatar-img emotion-3"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAABmJLR0QA/wD/AP+gvaeTAAAAq0lEQVRYhWPUSX77nwEPuLhDCYW/YQYTPuUMARn/UPj6HvfwqsdvGh3AqAMY/0rz400DhOKQEEBPQ+hgwENg1AGM6OUAoTijFKCnqQEPgVEHMK7dLIiSBtDLcnRAqFwglIbQ65IBD4FRB7CgCxCK48YAFbzy+gwE9DOg6h/wEBh1AEZdgA7Q8zWp5cBom3DQO4CF0jgmFaDrH/AQGHUAyW1CStPIaJtw0DkAAAhNNxMjfOg8AAAAAElFTkSuQmCC"
      />
    </div>
  </div>
  <div
    className="MuiCardHeader-content emotion-4"
  >
    <span
      className="MuiTypography-root MuiTypography-body2 MuiCardHeader-title emotion-5"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 emotion-6"
      >
        <div
          className="MuiGrid-root MuiGrid-item emotion-7"
        >
          <span
            className=" emotion-8"
          >
            <h5
              className="MuiTypography-root MuiTypography-h5  emotion-9"
              lines={3}
            >
              Auction
            </h5>
          </span>
        </div>
      </div>
      <div
        className="emotion-10"
      >
        <h2
          className="MuiTypography-root MuiTypography-h2  emotion-11"
          lines={3}
        >
          Market card with image icon
        </h2>
      </div>
    </span>
  </div>
</div>
`;

exports[`Snapshot testing MarketCardHeader Component renders correctly with image icon 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 16px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  margin-right: 16px;
}

.emotion-2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 40px;
  height: 40px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-size: 1.25rem;
  line-height: 1;
  border-radius: 50%;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-2.hasBackground {
  background-color: rgba(29, 34, 39, 0.04);
}

.emotion-2.xs {
  width: 24px;
  height: 24px;
}

.emotion-2.sm {
  width: 32px;
  height: 32px;
}

.emotion-2.md {
  width: 40px;
  height: 40px;
}

.emotion-2.lg {
  width: 48px;
  height: 48px;
}

.emotion-2.xl {
  width: 64px;
  height: 64px;
}

.emotion-2.xxl {
  width: 80px;
  height: 80px;
}

.emotion-2.max {
  width: 144px;
  height: 144px;
}

.emotion-3 {
  width: 100%;
  height: 100%;
  text-align: center;
  object-fit: cover;
  color: transparent;
  text-indent: 10000px;
}

.emotion-4 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-5 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  display: block;
}

.emotion-6 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: calc(100% + 8px);
  margin-top: -8px;
  width: calc(100% + 8px);
  margin-left: -8px;
  -webkit-box-pack: inherit;
  -ms-flex-pack: inherit;
  -webkit-justify-content: inherit;
  justify-content: inherit;
}

.emotion-6>.MuiGrid-item {
  padding-top: 8px;
}

.emotion-6>.MuiGrid-item {
  padding-left: 8px;
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-8 {
  border-radius: 0.2em;
  padding: 0.3em 0.6em;
  display: inline-block;
  background-color: #1976d2;
  color: #fff;
}

.emotion-8.label--icon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-8.label--icon>* {
  margin-right: 0.3em;
}

.emotion-8.label--icon>svg {
  width: 0.8em;
  height: 0.8em;
}

.emotion-9 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1.5rem;
  line-height: 1.334;
  letter-spacing: 0em;
}

.emotion-9.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.emotion-10 {
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  overflow: hidden;
  padding: 0.3em 0;
}

.emotion-11 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 300;
  font-size: 3.75rem;
  line-height: 1.2;
  letter-spacing: -0.00833em;
  font-weight: 700;
}

.emotion-11.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

<div
  className="MuiCardHeader-root emotion-0"
>
  <div
    className="MuiCardHeader-avatar emotion-1"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circular xl hasBackground emotion-2"
    >
      <img
        className="MuiAvatar-img emotion-3"
        src="https://w.wallhaven.cc/full/vg/wallhaven-vg7lv3.jpg"
      />
    </div>
  </div>
  <div
    className="MuiCardHeader-content emotion-4"
  >
    <span
      className="MuiTypography-root MuiTypography-body2 MuiCardHeader-title emotion-5"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 emotion-6"
      >
        <div
          className="MuiGrid-root MuiGrid-item emotion-7"
        >
          <span
            className=" emotion-8"
          >
            <h5
              className="MuiTypography-root MuiTypography-h5  emotion-9"
              lines={3}
            >
              Auction
            </h5>
          </span>
        </div>
      </div>
      <div
        className="emotion-10"
      >
        <h2
          className="MuiTypography-root MuiTypography-h2  emotion-11"
          lines={3}
        >
          Market card with image icon
        </h2>
      </div>
    </span>
  </div>
</div>
`;
